src/main/java/org/openjdk/jextract/JextractTool.java

```diff
@@ -241,13 +241,13 @@ OptionSet parse(String[] args) {
                    OptionSpec spec = optionSpecs.get(arg);
                    String argValue = null;
                    if (spec == null) {
-                       // check if we have usage like -C<clang_option> such as -C-xc++
-                       int idx = arg.indexOf('-', 1);
-                       if (idx != -1) {
-                           argValue = arg.substring(idx);
-                           spec = optionSpecs.get(arg.substring(0, idx));
-                       }
-                       if (spec == null) {
+                       // check for single char option specifier followed
```

Comment:
few suggestions:
* add a property to the option spec that says "I'm ok with no spaces"
* add a check in this logic which, before blindly splitting the argument string, it checks that it's one of the weird option specs with no space and required value
* it would be nice if, instead of hardwiring the 2, we could do something else, so that we can support even options with no spaces, but whose name is longer than 1 char